Jenkinsfile

```diff
@@ -613,6 +622,52 @@ def doBuildAppleDevice(String sdk, String buildType) {
     }
 }
 
+def doLinuxCrossCompile(String target, String buildType, String emulator = null) {
+    return {
+        node('docker') {
+            getArchive()
+            docker.build("realm-core-crosscompiling:${target}", "-f ${target}.Dockerfile .").inside {
+                try {
+                    dir('build-dir') {
+                        sh """
+                            cmake -GNinja \
+                                -DREALM_SKIP_SHARED_LIB=ON \
+                                -DCMAKE_TOOLCHAIN_FILE=$WORKSPACE/tools/cmake/${target}.toolchain.cmake \
+                                -DCMAKE_BUILD_TYPE=${buildType} \
+                                -DREALM_NO_TESTS=${emulator ? 'OFF' : 'ON'} \
+                                -DCPACK_SYSTEM_NAME=Linux-${target} \
+                                ..
+                        """
+
+                        runAndCollectWarnings(script: "ninja")
+
+                        if (emulator != null) {
+                            def environment = environment()
+                            environment << 'UNITTEST_PROGRESS=1'
+                            environment << 'UNITTEST_FILTER=- Thread_RobustMutex*'  // robust mutexes can't work under qemu
+                            withEnv(environment) {
+                                sh """
+                                    cd test
+                                    ulimit -s 256 # launching thousands of threads in 32-bit address space requires smaller stacks
+                                    LD_LIBRARY_PATH=/usr/arm-linux-gnueabihf/lib ${emulator} realm-tests
+                                """
+                            }
+                        } else {
+                            sh 'cpack'
+                                archiveArtifacts '*.tar.gz'
+                                def stashName = "linux-${target}___${buildType}"
+                                stash includes:"*.tar.gz", name:stashName
+                                publishingStashes << stashName
+                        }
+                    }
+                } finally {
+                    recordTests("Linux-${target}-${buildType}")
```

Comment:
The try/finally block should only wrap the actual running of the tests - everything inside `if (emulator != null)`. Otherwise Jenkins will try to record test results for all the package builds, not just the check builds.

Reply:
done